src/app/app.component.ts

```diff
@@ -23,13 +18,47 @@ export class AppComponent implements OnInit {
   private x_pos: number; // Stores x coordinate of the mouse pointer
   private x_gap: number; // Stores x gap (edge) between mouse and gutter
 
-  constructor(private endpointService: EndpointService,
-              private resourceService: ResourceService,
-              private transmartResourceService: TransmartResourceService,
-              private treeNodeService: TreeNodeService,
-              private constraintService: ConstraintService,
-              private queryService: QueryService,
-              private tableService: TableService) {
```

Comment:
these services are injected here to be accessible throughout the entire application as singular instances

Reply:
that's actually a mistake, will change back

Reply:
Wait - actually is it needed to be declared in AppComponent, if it is not used in AppComponent? Isn't it enough to be declared in the subcomponents or services only when they are used?
From my understanding as long as it is declared as a provider in the root module (AppModule), the injected instances will be singular, but I might be wrong.